packages/transaction-controller/src/TransactionController.ts

```diff
@@ -723,10 +764,12 @@ export class TransactionController extends BaseController<
       ]);
       transactionMeta.transactionHash = transactionHash;
       transactionMeta.status = TransactionStatus.submitted;
+      this.acceptApproval(transactionMeta);
```

Comment:
Any reason this is the only usage not done following all the existing logic? Shall we accept the approval after the update and event are emitted?

Reply:
It makes total sense, moved the updateTransaction.